specification/error-handling.md

```diff
@@ -52,7 +52,50 @@ Another example may be a metric exposed by a `SpanProcessor` that describes the
 Whenever the library suppresses an error that would otherwise have been exposed to the user, the library SHOULD log the error using language-specific conventions.
 SDKs MAY expose callbacks to allow end users to handle self-diagnostics separately from application code.
 
-## Exceptions to the rule
+## Configuring Error Handlers
 
-SDK authors MAY supply settings that allow end users to change the library's default error handling behavior.
+SDK authors MUST supply settings that allow end users to change the library's default error handling behavior.
 Application developers may want to run with strict error handling in a staging environment to catch invalid uses of the API, or malformed config.
+Note that configuring a custom error handler in this way is the only exception to the basic error handling principles outlined above. 
+
+The mechanism by which end users set or register a custom error handler should follow language-specific conventions. 
+SDK authors SHOULD provide a mechanism to distinguish between different types of errors within a custom error handler. For example, 
+an end user should have the ability to selectively ignore errors surfaced by the `SpanExporter` while logging all other errors. 
```

Comment:
Error types are good, but I predict we will end up being asked to standardize the error types, or at least a common subset of them. I don't know if defining them now would be required or helpful for the PR to be accepted, but it will probably come up as groups start to implement this. If we don't really know what the types are yet, perhaps that should be noted; either in the spec or in the PR message.

Reply:
I agree with Ted. I can imagine this exploding into a new sub topic. So let's go without the error types *for now*.

Reply:
Is it still worth including this 'SHOULD' section here, or should I remove this altogether?

Reply:
I'd probably be in favor of removing this because I'm not sure it adds much beyond what's already obvious / implied by the language